Automattic/wp-calypso
# Add `webpack-config-flag-plugin`.

A new `webpack` plugin, that safely replaces `config.isEnabled` flag checks with literals, taking into account the various possible ways `isEnabled` can be imported.

It's being used to replace just the `desktop` config flag, initially, as that's the same across all (web) environments.

This work evolved from @jsnajdr's excellent explorations in #31587.

#### Changes proposed in this Pull Request

* Add `webpack-config-flag-plugin`.
* Add extensive tests for plugin.
* Drive-by fixes to `state/lib/middleware` in preparation for using the plugin with the `desktop` flag. Taken from #31587.
* Replace `desktop` config flag using plugin.

#### Testing instructions

* Ensure that the build works normally
* Ensure that the `async-load-jquery` chunk is not being built
